functions/manager.py

```diff
@@ -423,32 +423,65 @@ def _runtask(body: Dict[str, Union[str, None]]) -> Boto3Result:
     Keys:
         entrypoint: If set, the entryPoint command field in the ECS task
         definition will be changed to this value before the task is started.
+
+        container_id: The container to run as a new task.
+
+        service_id: The service where the container can be found.
+
+        cluster_id: The cluster where the service can be found.
+
+    Returns:
+        Boto3Result
     """
-    taskdef_entrypoint = body.get("entrypoint")
+    taskdef_entrypoint: str = body.get("entrypoint") or ""
+    if not isinstance(taskdef_entrypoint, str):
+        err_msg = {
+            "msg": "TypeError",
+            "data": "'entrypoint' key must be of type string",
+            "level": "critical",
+        }
+        log(**err_msg)
+        return Boto3Result(exc=TypeError(err_msg))
+
+    missing_required_keys: List[Optional[str]] = []
+    required_keys = {"container_id", "service_id", "cluster_id"}
+    validated = {
+        key: value
+        for key in required_keys
+        for value in (body.get(key) or "",)
+        if value and isinstance(value, str)
+    }
+    missing_required_keys = sorted(required_keys - validated.keys())
+
+    if missing_required_keys:
+        err_msg = _missing_required_keys(list(required_keys), list(body))
+        log(**err_msg)
+        return Boto3Result(exc=KeyError(err_msg))
 
-    _environment = os.environ["ENVIRONMENT"]
-    _cluster = os.environ["ECS_CLUSTER"]
-    _service = os.environ["ECS_SERVICE"]
-    taskdef_family = f"{_service}-lambda-{_environment}"
+    container_id = validated["container_id"]
+    service_id = validated["service_id"]
+    cluster_id = validated["cluster_id"]
 
     ecs = boto3.client("ecs")
 
     r = invoke(
-        ecs.describe_services, **{"cluster": _cluster, "services": [_service]}
+        ecs.describe_services,
+        **{"cluster": cluster_id, "services": [service_id]},
     )
     if r.error:
         return r
     netconf = r.body["services"][0]["networkConfiguration"]
     svc_taskdef_arn = r.body["services"][0]["taskDefinition"]
 
     if taskdef_entrypoint:
-        _container_name = os.environ["ECS_CONTAINER"]
+        container_id = os.environ["ECS_CONTAINER"]
```

Comment:
```suggestion
```

Is this needed? There isn't an `ECS_CONTAINER` environment variable defined in the terraform. And this would overwrite the container_id that the caller passes in.